eng/targetingpacks.targets

```diff
@@ -164,9 +178,23 @@
           AfterTargets="ResolveTargetingPackAssets">
     <ItemGroup>
       <RuntimeFramework Condition="'%(RuntimeFramework.FrameworkName)' == '$(LocalFrameworkOverrideName)'">
-        <Version Condition="'$(StageTwoBuild)' != 'true'">$(ProductVersion)</Version>
-        <Version Condition="'$(StageTwoBuild)' == 'true'">$(BundledNETCoreAppPackageVersion)</Version>
+        <Version>$(ProductVersion)</Version>
       </RuntimeFramework>
     </ItemGroup>
   </Target>
+
+  <PropertyGroup Condition="'$(UseLocalNativeAotPack)' == 'true'">
+    <IlcToolsPath>$(CoreCLRCrossILCompilerDir)</IlcToolsPath>
+    <SysRoot Condition="('$(CrossBuild)' == 'true' or '$(BuildArchitecture)' != '$(TargetArchitecture)') and '$(HostOS)' != 'windows'">$(ROOTFS_DIR)</SysRoot>
```

Comment:
toolAot.targets line 38 sets SysRoot (basically overwrites it), do we need it here?

Reply:
Its needed here to support NativeAOTing the libraries tests (where this block came from).

There's likely more places we can clean up.

Reply:
If libraries tests include toolAot.targets, then technically it shouldn't need it. I agree that there are some mysteries around toolAot. e.g. `src/tests` requires these again on community platforms for some reason: https://github.com/am11/CrossRepoCITesting/blob/f327dfbd066ca551df71cda479935b9355c24342/.github/workflows/linux-riscv64-aot.yml#L48

```
SysRoot=$ROOTFS_DIR LinkerFlavor=lld /runtime/src/tests/build.sh \
  -riscv64 -cross -nativeaot -tree:nativeaot -Checked -p:TasksConfiguration=Debug \
  -p:LibrariesConfiguration=Debug -keepnativesymbols -p:UseLocalAppHostPack=true \
  -p:UseNativeAotForComponents=true -p:TestNativeAot=true
```
Would be nice if we don't have to specify `UseNativeAotForComponents` as property and `LinkerFlavor` & `SysRoot` as environment variables.

Reply:
I'd like to leave messing with the src/tests tree for later as that's always a mess and this PR already has a lot of nuance.